.github/workflows/build.yml

```diff
@@ -9,48 +9,82 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
-    - name: Install node
-      uses: actions/setup-node@v1
+      uses: actions/checkout@v3
+
+    - name: Base Setup
+      uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+    - name: Install dependencies
+      run: python -m pip install -U "jupyterlab>=4.0.0b0,<5"
+
+    - name: Lint the extension
+      run: |
+        set -eux
+        jlpm
+        jlpm run lint:check
+
+    - name: Build the extension
+      run: |
+        set -eux
+        python -m pip install .[test]
+
+        jupyter labextension list
+        jupyter labextension list 2>&1 | grep -ie "jupyterlab-skip-traceback.*OK"
+        python -m jupyterlab.browser_check
+
+    - name: Package the extension
+      run: |
+        set -eux
+
+        pip install build
+        python -m build
+        pip uninstall -y "jupyterlab-skip-traceback" jupyterlab
+
+    - name: Upload extension packages
```

Comment:
The artefacts just used in the `test_isolated` step. This could in principle be condensed into a single job, but it's the default for the `ts-extension-cookiecutter` repo, so I kept it. It's really up to you if you want to get rid of `test_isolated` :shrug: